compiler/src/lnntoamm/Expr.ts

```diff
@@ -595,6 +643,94 @@ class Const extends Expr {
   }
 }
 
+class New extends Expr {
+  valTy: Type
+  fields: {[name: string]: Ref}
+
+  get ty(): Type {
+    return this.valTy;
+  }
+
+  /**
+   * NOTE: does NOT check to make sure that the fields
+   * are valid.
```

Comment:
This note doesn't seem valid? I see checking below? Unless you mean the raw constructor does not check, but `fromTypeLiteral` does, in which case maybe a matching comment for that one? I thought this was for the whole class at first glance.

Reply:
Yeah, the checking is only done in `fromTypeLiteral` though - I was leaving this comment specifically for the constructor (I used JSDoc so that if somebody has to manually construct the `New`, they'll hopefully see it in their IDE as a reminder). I'll update the comment to note that